src/target/stm32f1.c

```diff
@@ -108,7 +108,10 @@ static void stm32f1_add_flash(target *t,
 	f->length = length;
 	f->blocksize = erasesize;
 	f->erase = stm32f1_flash_erase;
-	f->write = stm32f1_flash_write;
+	f->write = target_flash_write_buffered;
```

Comment:
Please add a comment explaining why this is needed with reference to the STM32F3 Reference Manual.  Also remove the `align` field, as it's not important when using the buffered model.